### PR TITLE
Add docker compose example incl. documentation

### PR DESCRIPTION
**Pull request by swvanbuuren**
This commit adds a functional docker compose example. It also comes with documentation that explains how to run, modify and use the provided example.

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        squad --bind 0.0.0.0:8000 --timeout=60 --workers=8 --log-level DEBUG
+    environment:
+      DATABASE: "ENGINE=django.db.backends.postgresql_psycopg2:NAME=squad:HOST=postgres:USER=squad:PASSWORD=squad"
+      SQUAD_STATIC_DIR: "/app/static"
+      SQUAD_EXTRA_SETTINGS: "/app/extra_settings.py"
+      SQUAD_CELERY_BROKER_URL: "amqp://rabbitmq"
+      SQUAD_BASE_URL: "https://www.your-squad-url.org"
+      # SENTRY_DSN: "<sentry-dsn>"
+      SECRET_KEY_FILE: "/app/secret.dat"
+      SQUAD_SITE_NAME: "CIP SQUAD Site"
```
Comment (chaws) [inline docker/docker-compose.yml:41]: Could you please leave it a generic name, just "SQUAD Site"?

**Comment by chaws** [comment]
This is super great! Thanks a bunch for putting this together.

I have one request though. Google chrome did not like port `10080`, it requires an extra setting in the browser to allow it accessing localhost on that port. I know we all run local servers so maybe hard coding to one specific local port isn't easy.

I propose the following patch:

```diff
diff --git a/docker/docker-compose.yml b/docker/docker-compose.yml
index e4ae6e7d..2b8b9492 100644
--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -138,7 +138,7 @@ services:
       - squad-static-files:/squad_static/static
       - ./squad-apache.conf:/usr/local/apache2/conf/httpd.conf
     ports:
-      - 10080:10080
+      - ${SQUAD_DOCKER_PORT:-10080}:80
 
 volumes:
   squad-static-files:
diff --git a/docker/squad-apache.conf b/docker/squad-apache.conf
index a2f497e4..ea9dbd85 100644
--- a/docker/squad-apache.conf
+++ b/docker/squad-apache.conf
@@ -1,6 +1,6 @@
 LogLevel debug
 
-Listen 10080
+Listen 80
 
 # Load some mods
 LoadModule headers_module modules/mod_headers.so
@@ -36,7 +36,7 @@ AddType image/svg+xml .svg
 
 ServerName www.your-squad-url.org
 
-<VirtualHost *:10080>
+<VirtualHost *:80>
       ProxyPreserveHost On
 
       RewriteEngine On
```

This way, it defaults to port 10080, but it is easy to change it, just `export SQUAD_DOCKER_PORT=8001` for instance.

File: docker/docker-compose.yml (hunk 1 above)
Comment (swvanbuuren) [reply docker/docker-compose.yml]: I missed this. Thanks for noting.

Changed with commit 149a5fa

**Comment by swvanbuuren** [comment]
> This is super great! Thanks a bunch for putting this together.
> 
> I have one request though. Google chrome did not like port `10080`, it requires an extra setting in the browser to allow it accessing localhost on that port. I know we all run local servers so maybe hard coding to one specific local port isn't easy.
> 
> I propose the following patch:
> 
> ```diff
> diff --git a/docker/docker-compose.yml b/docker/docker-compose.yml
> index e4ae6e7d..2b8b9492 100644
> --- a/docker/docker-compose.yml
> +++ b/docker/docker-compose.yml
> @@ -138,7 +138,7 @@ services:
>        - squad-static-files:/squad_static/static
>        - ./squad-apache.conf:/usr/local/apache2/conf/httpd.conf
>      ports:
> -      - 10080:10080
> +      - ${SQUAD_DOCKER_PORT:-10080}:80
> ...
> ```
> 
> This way, it defaults to port 10080, but it is easy to change it, just `export SQUAD_DOCKER_PORT=8001` for instance.

Yes, I can see this might cause problems for other users. I have taken over your suggestion in commit [cb11918](https://github.com/Linaro/squad/pull/1121/commits/cb11918f2cf96496d1a27620cf7a91d667d4f96d)

**Comment by swvanbuuren** [comment]
@chaws I have made all the changes you've requested/suggested.

Are there any other open points?

**Review by chaws — APPROVED** [review]
LGTM

**Comment by chaws** [comment]
> @chaws I have made all the changes you've requested/suggested.
> 
> Are there any other open points?

Hi @swvanbuuren thanks for fixing it!

**Pull request merged**